packages/cubejs-postgres-driver/driver/PostgresDriver.js

```diff
@@ -51,6 +51,26 @@ class PostgresDriver extends BaseDriver {
     }
   }
 
+  async uploadTable(table, columns, tableData) {
+    if (!tableData.rows) {
+      throw new Error(`${this.constructor} driver supports only rows upload`);
+    }
+    await this.createTable(table, columns);
+    try {
+      const q = `INSERT INTO ${table}
+      (${columns.map(c => this.quoteIdentifier(c.name)).join(', ')})
+      SELECT * FROM UNNEST (${columns.map((c, columnIndex) => `${this.param(columnIndex)}::${this.toGenericType(c.type)}[]`).join(', ')})`;
```

Comment:
Could you please elaborate on scenario where this cast is required? Also is it `toGenericType` and not `fromGenericType`?

Reply:
Hmm, `toGenericType` is redundant here, you're right. 
Now I see that types are being casted in `createTableSql` method.